packages/react/toast/src/Toast.tsx

```diff
@@ -93,7 +93,13 @@ const ToastProvider: React.FC<ToastProviderProps> = (props: ScopedProps<ToastPro
         toastCount={toastCount}
         viewport={viewport}
         onViewportChange={setViewport}
-        onToastAdd={React.useCallback(() => setToastCount((prevCount) => prevCount + 1), [])}
+        onToastAdd={React.useCallback(() => {
+          // Reset the isClosePausedRef when a new toast is added
+          isClosePausedRef.current = false;
+          // I assume this one has the same issue as the one above
+          // isFocusedToastEscapeKeyDownRef.current = false;
```

Comment:
I assume the `isFocusedToastEscapeKeyDownRef` has the same issue of not initializing with `false` between rerenders, so we might need to reset that too